crates/sui-core/src/authority_active.rs

```diff
@@ -231,11 +224,18 @@ where
             }
         });
 
-        if let Err(err) = _gossip_join.await {
-            error!("Join gossip task end error: {:?}", err);
-        }
         if let Err(err) = _checkpoint_join.await {
             error!("Join checkpoint task end error: {:?}", err);
         }
     }
+
+    /// Spawn gossip process
+    pub async fn spawn_gossip_process(self, degree: usize) -> JoinHandle<()> {
```

Comment:
If both spawn_checkpoint_process and spawn_gossip_process takes `self` as argument, how would we start both processes?

Reply:
Good point. When we are ready to start the checkpoint process as well, we can re-combine these into a single function.